core/src/main/scala/pureconfig/ConfigConvert.scala

```diff
@@ -235,6 +235,23 @@ object ConfigConvert extends LowPriorityConfigConvertImplicits {
           }
 
           tryBuilder.map(_.result())
+        case o: ConfigObject =>
```

Comment:
@melrief 
Good point about `Throwable`. What I should've used is `NonFatal`. The updated function would look like:
```
    private def reportBadIndex(keyString: String): PartialFunction[Throwable, Failure[Nothing]] = {
      case scala.util.control.NonFatal(e) =>
        val message = s"Cannot interpet $keyString as a numeric index. Tried to read the object as an indexed sequence. " +
          "Expecting syntax like (a.0=0, a.1=1, ...) when loading a sequence called 'a'"
        Failure(new IllegalArgumentException(message, e))
    }
```